start.py

```diff
@@ -48,35 +56,54 @@ def main(args):
     secondary_nodes = [Node(hostname=hostname, group='secondary', image=secondary_node_image)
                        for hostname in args.secondary_nodes]
 
-    cluster = Cluster(primary_node, *secondary_nodes)
+    cluster = Cluster(primary_node, *secondary_nodes, name=args.cluster_name)
     cluster.primary_node = primary_node
     cluster.secondary_nodes = secondary_nodes
     cluster.start(args.network)
 
+    hdp_version_tuple = version_tuple(args.hdp_version)
+
     logger.debug('Starting PostgreSQL for Ambari server ...')
-    primary_node.execute('service postgresql start', quiet=not args.verbose)
-    _update_node_names(cluster, quiet=not args.verbose)
+    # Need this as init system in Docker misreports on postgres start initially
+    # Check https://github.com/docker-library/postgres/issues/146 for more
+    def condition():
+        if '1 row' not in primary_node.execute('PGPASSWORD=bigdata psql ambari '
```

Comment:
Just a thought: you may just do a check for the affirmative case (`if '1 row' in ...`) and then return `True` after restarting `postgresql`. If you leave out the `return False`, it keeps looping implicitly because `bool(None) == False` as far as Python is concerned.